### PR TITLE
=cdd #17778 Convert samples to java

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #17778

File: akka-samples/akka-sample-distributed-data-java/COPYING [hunk 1]
```diff
@@ -0,0 +1,121 @@
+Creative Commons Legal Code
+
+CC0 1.0 Universal
```
Comment (ktoso) [inline akka-samples/akka-sample-distributed-data-java/COPYING]: I don't mind, but is this by purpose or by accident?

Comment (ktoso) [inline akka-samples/akka-sample-distributed-data-java/COPYING]: (rest of akka is apache 2)

File: akka-samples/akka-sample-distributed-data-java/LICENSE [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Activator Template by Typesafe
+
+Licensed under Public Domain (CC0)
+
+To the extent possible under law, the person who associated CC0 with
+this Activator Tempate has waived all copyright and related or neighboring
+rights to this Activator Template.
+
+You should have received a copy of the CC0 legalcode along with this
+work.  If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
```
Comment (ktoso) [inline akka-samples/akka-sample-distributed-data-java/LICENSE]: Oh, so I guess activator generates those nowadays?

File: akka-samples/akka-sample-distributed-data-java/src/main/java/sample/distributeddata/ReplicatedMetrics.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
+package sample.distributeddata;
```
Comment (ktoso) [inline akka-samples/akka-sample-distributed-data-java/src/main/java/sample/distributeddata/ReplicatedMetrics.java]: Sanity check, no license header needed?

File: akka-samples/akka-sample-distributed-data-java/tutorial/index.html [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+merge function, which all data types must provide. The state changes always converge.
+Several useful data types for counters, sets, maps and registers are provided and
+you can also implement your own custom data types.
+</p>
+
+<p>
+It is eventually consistent and geared toward providing high read and write availability
+(partition tolerance), with low latency. Note that in an eventually consistent system a read may return an 
+out-of-date value.
+</p>
```
Comment (ktoso) [inline akka-samples/akka-sample-distributed-data-java/tutorial/index.html]: Should you point out that it is for "small data"?

**Comment by ktoso** [comment]
LGTM, curious if CC0 choice was conscious or not (I don't mind the license, it's fine for samples)

File: akka-samples/akka-sample-distributed-data-java/COPYING (hunk 1 above)
Comment (patriknw) [inline akka-samples/akka-sample-distributed-data-java/COPYING]: all samples (activator templates) have this

File: akka-samples/akka-sample-distributed-data-java/src/main/java/sample/distributeddata/ReplicatedMetrics.java (hunk 3 above)
Comment (patriknw) [inline akka-samples/akka-sample-distributed-data-java/src/main/java/sample/distributeddata/ReplicatedMetrics.java]: the reason is that we don't want to scare people by typesafe copyrights in samples

File: akka-samples/akka-sample-distributed-data-java/tutorial/index.html (hunk 4 above)
Comment (patriknw) [inline akka-samples/akka-sample-distributed-data-java/tutorial/index.html]: yes, perhaps a link to http://doc.akka.io/docs/akka/2.4-M2/scala/distributed-data.html#Limitations

File: akka-samples/akka-sample-distributed-data-java/src/main/java/sample/distributeddata/ReplicatedMetrics.java (hunk 3 above)
Comment (ktoso) [inline akka-samples/akka-sample-distributed-data-java/src/main/java/sample/distributeddata/ReplicatedMetrics.java]: Yeah, completely makes sense. Somehow didn't notice this before in others

File: akka-samples/akka-sample-distributed-data-java/tutorial/index.html (hunk 4 above)
Comment (ktoso) [inline akka-samples/akka-sample-distributed-data-java/tutorial/index.html]: yes please :)

**Pull request merged**